src/merkle_tree/mod.rs

```diff
@@ -0,0 +1,184 @@
+use std::ops::Rem;
+
+use crate::{hash, hash_pair};
+
+pub struct MerkleTree {
+    hashes: Vec<Vec<u8>>,
+    count: usize
+}
+
+impl MerkleTree {
+    pub fn new(elements: &[String]) -> MerkleTree{
+        let mut tree = MerkleTree{hashes: vec![], count:  elements.len()};
+        let leaves = tree.hash_elements(elements);
+        tree.hashes = tree.build_hashes(leaves);
+        tree
+    }
+
+    pub fn root(&self) -> &[u8] {
+        &self.hashes[0]
+    }
+
+    pub fn add(&mut self, elements: &[String]) {
+        let new_leaves = self.hash_elements(&elements);
+        let old_leaves = self.hashes[{self.count-1}..].to_vec();
+        let leaves = [old_leaves, new_leaves].concat();
+        let count =  leaves.len();
+        self.hashes = self.build_hashes(leaves);
+        self.count = count;
+    }
+
+    fn hash_elements(&self, elements: &[String]) -> Vec<Vec<u8>> {
+        elements.iter().map(|e| hash(e.to_string())).collect()
+    }
+
+    fn build_hashes(&self, hashes: Vec<Vec<u8>>) -> Vec<Vec<u8>> {
```

Comment:
The same as with `hash_elements`: this could be `self`-less. Dropping it allows you to keep `MerkleTree::new` a bit cleaner:
```rust
pub fn new(elements: &[String]) -> MerkleTree {
    let leaves = Self::hash_elements(elements.clone());
    let hashes = Self::build_hashes(leaves);
    MerkleTree { hashes, count: elements.len() }
}
```